### ISSUE TITLE
Retrieve auth token from git credential helper

### ISSUE THREAD

**Issue by oyamauchi** [getcord/spr#83]
Git has a way to [store repo credentials](https://git-scm.com/docs/gitcredentials), which includes GitHub tokens; spr should have a way to use that, instead of needing `spr.githubAuthToken`. This would avoid having copies of tokens all over the place in `.git/config` files.

In my global .gitconfig:

```
[credential]
helper = store
```

The `store` helper by default stores credentials in `~/.git-credentials`, which looks like:

```
https://oyamauchi:ghp_*******@github.com
```

Then this code retrieves credentials for the repo at the URL you pass to `CredentialHelper::new`:

```rust
use git2::Config;
use git2::CredentialHelper;

fn main() {
  let config = Config::open_default().unwrap();
  let mut cred_helper = CredentialHelper::new("https://github.com/getcord/spr.git");
  cred_helper.config(&config);
  println!("{:?}", cred_helper.execute());
}
```

Which outputs:

```
Some(("oyamauchi", "ghp_*******"))
```

**Comment by oyamauchi** [comment]
Yeah, I don't think it necessarily has to be git-credentials, just some kind of centralized storage. (1) because of the security angle, and (2) because it's more convenient, like if you have multiple repos where you're using spr.

**Comment by sunshowers** [comment]
Related to this, it would be great if we could configure the token to be fetched via a simple command -- my personal preference for this would be to store it in 1password and use their shell command `op` to fetch it.
